proto/buf/alpha/registry/v1alpha1/reference.proto

```diff
@@ -16,29 +16,35 @@ syntax = "proto3";
 
 package buf.alpha.registry.v1alpha1;
 
-import "buf/alpha/registry/v1alpha1/repository_branch.proto";
 import "buf/alpha/registry/v1alpha1/repository_commit.proto";
 import "buf/alpha/registry/v1alpha1/repository_tag.proto";
-import "buf/alpha/registry/v1alpha1/repository_track.proto";
 
 message Reference {
+  reserved 1, 4;
+  reserved "branch", "track";
+
   oneof reference {
-    // The requested reference is a branch.
-    RepositoryBranch branch = 1;
     // The requested reference is a tag.
     RepositoryTag tag = 2;
     // The requested reference is a commit.
     RepositoryCommit commit = 3;
-    // The requested reference is a track.
-    RepositoryTrack track = 4;
+    // The requested reference is the default reference.
+    RepositoryMainReference main = 5;
```

Comment:
@seankimdev @timostamm related to FE-154, my understanding is that we'd need something like this if the `ReferenceService` continues to exist. The follow up pr in https://github.com/bufbuild/buf/pull/1219/files#diff-9c96176c5e8386f22fe9a9f8f8e32bd0dff996b855ec337245229311eda5508eR34 shows how drafts could be stacked on top of this.

Reply:
I think that makes sense. @timostamm, what do you think?